theano/sandbox/gpuarray/elemwise.py

```diff
@@ -576,8 +604,8 @@ class GpuCAReduceCuda(HideC, CAReduceDtype):
 
     pre_scalar_op: if present, must be a scalar op with only 1
     input. We will execute it on the input value before reduction.
-
     """
+    _f16_ok = True
```

Comment:
DimShuffle does not need any change at the C level since it doesn't compute anything with the data.